test/core/mfs-preload.spec.js

```diff
@@ -41,16 +43,17 @@ describe('MFS preload', () => {
     preloader.start((err) => {
       expect(err).to.not.exist()
 
-      setTimeout(() => {
-        preloader.stop((err) => {
-          expect(err).to.not.exist()
-          expect(
-            // Slice off any extra CIDs it processed
-            mockPreload.cids.slice(0, expectedPreloadCids.length)
-          ).to.deep.equal(expectedPreloadCids)
-          done()
-        })
-      }, statCids.length * (interval * 2))
+      const test = (cb) => {
+        // Slice off any extra CIDs it processed
+        const cids = mockPreload.cids.slice(0, expectedPreloadCids.length)
+        if (cids.length !== expectedPreloadCids.length) return cb(null, false)
+        cb(null, cids.every((cid, i) => cid === expectedPreloadCids[i]))
+      }
+
+      waitFor(test, { name: 'CIDs to be preloaded' }, (err) => {
+        expect(err).to.not.exist()
```

Comment:
rather than `expect(err).to.not.exist()` we should be using `if (err) return done(err)` so we properly finish off the test when it errors

Reply:
Do you mean so that we always call `preloader.stop`?

I'm not sure I understand why using `done(err)` is necessary in this case and not elsewhere in the tests?

Reply:
ping @VictorBjelkholm